### DISCUSSION TITLE
Routing not working as expected when using object with Link component

### DISCUSSION THREAD

**Discussion by sirmyron** [vercel/next.js#26007] [Help]
### What version of Next.js are you using?

10.2.3

### What version of Node.js are you using?

16.1.0 

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

next start

### Describe the Bug

Using an [object](https://nextjs.org/docs/api-reference/next/link#with-url-object) AND passing query parameters to the Link component, my do a full refresh instead of being routed as expected.  The urls get build correctly in all cases, but it just does not route correctly. 

Something I'm doing incorrectly?

### Expected Behavior

I expect my links to be routed correctly through nextjs and not do a full page refresh. 

The same way these examples work as expected:

```
<Link href={{ pathname: `/reports/${id}/${slug}` } }}>
    <a className="text-blue-500">{reportItem.title}</a>
</Link>
```

```
<Link href={`/reports/${id}/${slug}`}>
    <a className="text-blue-500">{reportItem.title}</a>
</Link>
```

### To Reproduce

The following examples don't work:

```
<Link href={{ pathname: `/reports/[id]/[slug]`, query: { id, slug } }}>
    <a className="text-blue-500">{reportItem.title}</a>
</Link>
```

```
<Link href={{ pathname: `/reports/[id]/${slug}`, query: { id } }}>
    <a className="text-blue-500">{reportItem.title}</a>
</Link>
```

**Comment by timneutkens** [comment]
The reproduction examples are not working because you're not interpolating the `id` and `slug`:
```
<Link href={{ pathname: `/reports/${id}/${slug}`, query: { myQuery: '1' } }}>
    <a className="text-blue-500">{reportItem.title}</a>
</Link>
```

Passing `pathname` with a Next.js directory path and query holding the values is not supported. If you're providing it in that way you have to use `as` but that's not needed in general.
